### PR TITLE
IGAPP-686: Migrate app module

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

I deleted all already migrated or not anymore needed files and moved everything else to web-ts. The only really relevant change is the migration of `initSentry` and the use of it in `App`.

vscode settings will be adjusted here: https://issues.tuerantuer.org/browse/IGAPP-688
documenation will be adjusted here: https://issues.tuerantuer.org/browse/IGAPP-598

### CODE REVIEW

**Review by Taggotty — APPROVED** [review]
Renaming of web happens in another PR? Or do we even want to rename web-ts to web?

**Comment by klinzo** [comment]
> Renaming of web happens in another PR? Or do we even want to rename web-ts to web?

I would rename it because the `-ts` suffix has no valuable information in the future. There is already an issue for it: https://issues.tuerantuer.org/browse/IGAPP-690

**Comment by klinzo** [comment]
@maxammann I was not able to send an error to sentry, could you perhaps test this please?

File: web-ts/tsconfig.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "isolatedModules": true,
     "jsx": "react",
     "lib": ["dom", "dom.iterable", "ES5", "ES6"],
+    "module": "ES2020",
```
Comment (maxammann) [inline web-ts/tsconfig.json:10]: This could be the reason why sentry is no longer working as sentry is loaded in a deferred way.

**Comment by maxammann** [comment]
> @maxammann I was not able to send an error to sentry, could you perhaps test this please?

I think the migration to TS is responsible that sentry is no longer working. Did you test whether sentry works before this PR?

**Comment by maxammann** [comment]
I'll test this soon.

**Comment by maxammann** [comment]
Sentry still works for me. I did a release build and disabled UBlock origin. Then it works :)

**Review by maxammann — APPROVED** [review]

**Pull request merged**